pkg/storage/gcs/store.go

```diff
@@ -125,26 +150,35 @@ func (g *gcs) GetAt(ctx context.Context, objectName string) (io.ReaderAt, error)
 	return &gcsReader{
 		g:          g,
 		objectName: objectName,
+		l:          g.l,
 	}, nil
 }
 
 func (g *gcs) Touch(ctx context.Context, objectName string) error {
+	g.l.Debug("Start Touch", zap.String("objectName", objectName))
 	_, err := g.client.Bucket(g.bucket).Object(objectName).Update(ctx, gcsStorage.ObjectAttrsToUpdate{})
+	g.l.Debug("End touch", zap.String("objectName", objectName))
 	return toSentinelErrors(err)
 }
 
 type readCloser struct {
 	reader io.Reader
 }
 
-func (rc readCloser) Read(p []byte) (n int, err error) {
+func (rc readCloser) Read(p []byte) (int, error) {
 	return rc.reader.Read(p)
 }
+
 func (rc readCloser) Close() error {
+	// TODO(fred): here is what the google storage api states on Write:
```

Comment:
This is for a read closer.. if the read completed then close can fail. This is not used for writes.

Reply:
ok